src/v/cloud_storage_clients/types.h

```diff
@@ -62,6 +63,17 @@ inline std::error_code make_error_code(error_outcome e) noexcept {
     return {static_cast<int>(e), error_category()};
 }
 
+enum class s3_url_style { virtual_host = 0, path };
+
+inline std::ostream& operator<<(std::ostream& os, const s3_url_style& us) {
+    switch (us) {
+    case s3_url_style::virtual_host:
+        return os << "virtual_host";
+    case s3_url_style::path:
+        return os << "path";
+    }
```

Comment:
can you stick the implementation in a .cc file?

Reply:
It's a bit tricky, because the `config` library is going to depend on this implementation. Having it `inline` and accessible from the header saves us from having to depend on the `cloud_storage_clients` library (and adding this would create a cyclic dependency).

Reply:
ahh, yeh, then don't worry about it. we'll fix those dependencies as we are switching modules over to be new-style. it's a lot of work no need to do that here.